spark-bigtable_2.12/src/main/scala/com/google/cloud/spark/bigtable/BigtableRDD.scala

```diff
@@ -0,0 +1,64 @@
+package com.google.cloud.spark.bigtable
+
+import com.google.cloud.bigtable.data.v2.models.{RowMutationEntry, Row => BigtableRow}
+import com.google.cloud.spark.bigtable.datasources.{BigtableClientKey, BigtableDataClientBuilder, BigtableTableScanRDD}
+import com.google.cloud.spark.bigtable.filters.RowKeyWrapper
+import com.google.common.collect.{ImmutableRangeSet, Range}
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import scala.collection.JavaConverters._
+
+class BigtableRDD(@transient val sparkContext: SparkContext) extends Serializable with Logging {
+
+  // Overloading the method to support both Scala and Java Maps
+  def readRDD(
+      tableId: String,
+      parameters: java.util.Map[String, String]
```

Comment:
I'm a little worried that we're just gonna let people pass in a map of random things. Can we create a BigtableOptions for setting these parameters?

Reply:
Do you mean a separate class with each of the fields we support?

The original reason we started using a Map[String, String] for DataFrames is that this is how Spark passes down options to the connector, [like here](https://github.com/GoogleCloudDataproc/spark-bigtable-connector/blob/83b92e35842e339d0e6d8e2a5a90edd012868061/examples/java-maven/src/main/java/spark/bigtable/example/WordCount.java#L129-L134), but we only check for parameters that are defined in side [BigtableSparkConf](https://github.com/GoogleCloudDataproc/spark-bigtable-connector/blob/main/spark-bigtable_2.12/src/main/scala/com/google/cloud/spark/bigtable/datasources/BigtableSparkConf.scala)

Since we're using the same infrastructure as DataFrames for RDDs, I also kept the same Map object for the parameters for consistency. We can change it and use a class instead, but I think it might make it a bit more complicated since that way we'll have two different ways to pass options between DataFrames and RDDs.

Reply:
Discussed offline, will submit the PR to get the periodic tests run over it, will double check the parameters details on Monday.